node/src/cli.rs

```diff
@@ -109,3 +112,42 @@ impl RelayChainCli {
         }
     }
 }
+
+#[derive(Debug, clap::Parser)]
+#[group(skip)]
+pub struct RunCmd {
+    #[clap(flatten)]
+    pub base: cumulus_client_cli::RunCmd,
+
+    /// P2P ED25519 private key
+    #[arg(long, value_parser = keypair_value_parser)]
+    pub p2p_key: Option<Keypair>,
+
+    /// Listen address that the bootstrap node binds to.
+    #[arg(long)]
+    pub p2p_listen_address: Option<Multiaddr>,
+}
+
+impl std::ops::Deref for RunCmd {
+    type Target = cumulus_client_cli::RunCmd;
+
+    fn deref(&self) -> &Self::Target {
+        &self.base
+    }
+}
+
+/// Parses a ED25519 private key into a Keypair.
+/// Takes in a private key or the path to a PEM file, depending on the @ prefix.
+pub(crate) fn keypair_value_parser(src: &str) -> Result<Keypair, String> {
```

Comment:
We use similar functions across the codebase, should we put somewhere and reuse it?